src/main/java/org/spongepowered/api/util/blockray/BlockRay.java

```diff
@@ -97,8 +97,10 @@
     private static final Vector3d Z_POSITIVE = Vector3d.UNIT_Z;
     private static final Vector3d Z_NEGATIVE = Z_POSITIVE.negate();
     private static final int DEFAULT_BLOCK_LIMIT = 1000;
-    // Ending test predicate
+    // Filter for removing hits
     private final Predicate<BlockRayHit<E>> filter;
+    // Ending test predicate
+    private final Predicate<BlockRayHit<E>> endPredicate;
```

Comment:
An `endPredicate` implies an inverted logic, compared to what it was previously. This naming is confusing.

Instead we should have into two kinds of filters. A "stopping" one, which is the current filter implementation, and a "skipping" one, which is what this PR is adding. The filters should be name `stopOn` and `skipOn`. A block that is reject by such a filter should never be returned by the iterator.